Make auto-conversion of response bodies sensitive to Content-Type header

edwardsph opened the pull request:
### Description

Presently, any response body that starts with `<` is auto-converted to XML but if it is not XML there is still a fatal error reported. This PR attempts to avoid that by checking the resource type before converting. It tries to avoid breaking existing behaviour by ensuring any response body that is of an ambiguous type will be handled as it was previously.

The resource type is passed (when available) into the JsValue converter. There is then a test to see if there is a ResourceType defined and if it is explicitly not a known JSON or XML (or plain text) format, then auto conversion is not attempted. I added a number of RDF media types that may start with a `<` but cannot be parsed as XML.

If you are happy with this approach, it could allow for the optional addition in the future of an RDFUtils class to convert any RDF types to triples. I wanted to take this one step at a time since you are rightly concerned not to break existing tests. 
One thing to note is that there are other XML conversions where an HTTP response and content type is not present. I suspect that if RDF were to be handled in these places too, then we would need a way to improve the isXml function to do more than check the first character. You may decide that is a step too far but this I think this PR still has value on its own as it will prevent the incorrect fatal error messages.

All tests pass but I wasn't sure where to add some additional tests for this change. If you can point me in the right direction I can add them to the PR.

- Relevant Issues : #1462 
- Type of change :
  - [ ] New feature
  - [x] Bug fix for existing feature
  - [ ] Code quality improvement
  - [ ] Addition or Improvement of tests
  - [ ] Addition or Improvement of documentation

ptrthomas commented:
@edwardsph this is good, don't mind merging - but as you said, good to have some tests. I think `KarateMockHandlerTest` may be the easiest place, you can script the (mock) reponses and the client side in one flow and test that that hidden-variable `responseType` is set as expected etc. let me know if you have questions on that

edwardsph commented:
I've added a few tests but also changed one so I think I need to explain a bit.

The `testResponseContentTypeForJson` created the response body by using Karate's JSON handling instead of relying on the content-type header so I modified this test and added a similar test for XML responses: `testResponseContentTypeForXml`.

There are 2 new tests to confirm that XML & JSON with incorrect content-types are still converted:
- `testResponseAutoConversionForXmlAsPlainText`
- `testResponseAutoConversionForJsonAsPlainText`

Finally I added 2 new tests for text that was previously autoconverted to XML even when it wasn't:
- `testResponseAutoConversionForTextWithTags` - exhibits old behaviour and shows a Fatal error
- `testResponseContentTypeForNonXmlWithTags` - uses the content-type to avoid unnecessary conversion

I hope that is all OK.
